presto-hive/src/test/java/io/prestosql/plugin/hive/AbstractTestHive.java

```diff
@@ -1722,6 +1724,71 @@ private static void assertBucketTableEvolutionResult(MaterializedResult result,
         assertEquals(idCount.keySet(), expectedIds);
     }
 
+    @Test
+    public void testBucketedTableValidation()
+            throws Exception
+    {
+        for (HiveStorageFormat storageFormat : createTableFormats) {
+            SchemaTableName table = temporaryTable("bucket_validation");
+            try {
+                doTestBucketedTableValidation(storageFormat, table);
+            }
+            finally {
+                dropTable(table);
+            }
+        }
+    }
+
+    private void doTestBucketedTableValidation(HiveStorageFormat storageFormat, SchemaTableName tableName)
+            throws Exception
+    {
+        createEmptyTable(
+                tableName,
+                storageFormat,
+                ImmutableList.of(
+                        new Column("id", HIVE_LONG, Optional.empty()),
+                        new Column("name", HIVE_STRING, Optional.empty())),
+                ImmutableList.of(),
+                Optional.of(new HiveBucketProperty(ImmutableList.of("id"), BUCKETING_V1, 8, ImmutableList.of())));
+
+        MaterializedResult.Builder dataBuilder = MaterializedResult.resultBuilder(SESSION, BIGINT, VARCHAR);
+        for (long id = 0; id < 100; id++) {
+            dataBuilder.row(id, String.valueOf(id));
+        }
+        insertData(tableName, dataBuilder.build());
+
+        try (Transaction transaction = newTransaction()) {
+            Set<String> files = listAllDataFiles(transaction, tableName.getSchemaName(), tableName.getTableName());
+
+            Path bucket2 = files.stream()
+                    .map(Path::new)
+                    .filter(path -> path.getName().startsWith("000002_0_"))
+                    .collect(onlyElement());
+
+            Path bucket5 = files.stream()
+                    .map(Path::new)
+                    .filter(path -> path.getName().startsWith("000005_0_"))
+                    .collect(onlyElement());
+
+            HdfsContext context = new HdfsContext(newSession(), tableName.getSchemaName(), tableName.getTableName());
+            FileSystem fileSystem = hdfsEnvironment.getFileSystem(context, bucket2);
+            fileSystem.delete(bucket2, false);
+            fileSystem.rename(bucket5, bucket2);
+        }
+
+        try (Transaction transaction = newTransaction()) {
```

Comment:
Add test validation that validation does not happen if it is disabled.

Reply:
Done